### PR TITLE
short term fix for not including dashes

### PR DESCRIPTION
**Pull request by emord**
@millerdev buddy @czue http://manage.dimagi.com/default.asp?239638

This will solve the issue of not being able to retrieve already saved attachments, but I will need to look deeper to make sure this is consistent between all forms. Offlined with @millerdev and it seems that ideally we would always use without dashes as it saves RAM

### CODE REVIEW

File: corehq/form_processor/models.py [hunk 1]
```diff
@@ -440,12 +444,15 @@ def delete_content(self):
...
-    def blobdb_bucket(self):
+    def blobdb_bucket(self, remove_dashes=False):
         if self.blob_bucket is not None:
             return self.blob_bucket
         if self.attachment_id is None:
             raise AttachmentNotFound("cannot manipulate attachment on unidentified document")
-        return os.path.join(self._attachment_prefix, str(self.attachment_id))
+        attach_id = str(self.attachment_id)
+        if remove_dashes:
+            attach_id = attach_id.replace('-', '')
```
Comment (millerdev) [inline corehq/form_processor/models.py]: You can get this with `self.attachment_id.hex` so
```py
if remove_dashes:
    attach_id = self.attachment_id.hex
else:
    attach_id = str(self.attachment_id)
```

File: corehq/form_processor/models.py [hunk 2]
```diff
@@ -423,7 +423,11 @@ def read_content(self, stream=False):
         try:
             blob = db.get(self.blob_id, self.blobdb_bucket())
         except (KeyError, NotFound, BadName):
-            raise AttachmentNotFound(self.name)
+            try:
+                # http://manage.dimagi.com/default.asp?239638
+                blob = db.get(self.blob_id, self.blobdb_bucket(True))
```
Comment (millerdev) [inline corehq/form_processor/models.py]: might want to check if `self.blob_bucket` is empty and `'-' in str(self.attachment_id)` before hitting riak again

File: corehq/form_processor/models.py [hunk 3]
```diff
@@ -423,7 +423,14 @@ def read_content(self, stream=False):
         try:
             blob = db.get(self.blob_id, self.blobdb_bucket())
         except (KeyError, NotFound, BadName):
-            raise AttachmentNotFound(self.name)
+            if not self.blob_bucket and '-' in str(self.attachment_id):
+                try:
+                    # http://manage.dimagi.com/default.asp?239638
+                    blob = db.get(self.blob_id, self.blobdb_bucket(True))
```
Comment (czue) [inline corehq/form_processor/models.py]: super nit-picky but for booleans i find it's nice to always call it as a kwarg so it's clear what the value is doing

**Review by czue — COMMENTED** [review]
no blockers from me - will let @millerdev formally approve/merge

**Review by millerdev — APPROVED** [review]

**Pull request merged**